Added parser and logger

animesh-007 opened the pull request:
Added parser and logger. Can you review it and let me know if there are any other changes that need to be made? I also wanted to ask what are things you would like to log (like models architect, metrics) using logger?

animesh-007 committed:
added parser and changed code according

animesh-007 committed:
removed extra comments code

animesh-007 committed:
removed extra code

animesh-007 committed:
solved issue in train_images.py

adityac8 commented:
Model arch logging is not required.
Metrics needs logging.

adityac8 commented on datasets/image_util.py:
```diff
@@ -3,9 +3,8 @@
-import sys
-import pdb
-import h5py
+# import sys
```
Remove them

adityac8 commented on datasets/image_util.py:
```diff
@@ -44,40 +43,41 @@ def read_matdataset(self, opt):
-                self.test_seen_feature = torch.from_numpy(_test_seen_feature).float() 
-                self.test_seen_feature.mul_(1/mx)
-                self.test_seen_label = torch.from_numpy(label[test_seen_loc]).long()
+        # if not opt.validation:
```
Seems ok, remove them.

adityac8 commented on datasets/image_util.py:
```diff
@@ -44,40 +43,41 @@ def read_matdataset(self, opt):
-            self.train_label = torch.from_numpy(label[train_loc]).long()
-            self.test_unseen_feature = torch.from_numpy(feature[val_unseen_loc]).float()
-            self.test_unseen_label = torch.from_numpy(label[val_unseen_loc]).long() 
+        # else:
```
These lines also removed.

adityac8 commented on networks/TFVAEGAN_model.py:
```diff
@@ -20,8 +20,9 @@ class Encoder(nn.Module):
         super(Encoder,self).__init__()
-        layer_sizes = opt.encoder_layer_sizes
-        latent_size = opt.latent_size
+        layer_sizes = [8192, 4096]
```
Don't put any numbers manually.

adityac8 commented on networks/TFVAEGAN_model.py:
```diff
@@ -45,8 +46,9 @@ def __init__(self, opt):
 
-        layer_sizes = opt.decoder_layer_sizes
-        latent_size=opt.latent_size
+        layer_sizes = [4096, 8192]
```
Don't put any numbers manually.

adityac8 commented on options/tfVAEGAN/CUB.yml:
```diff
@@ -7,23 +7,23 @@ datasets:
   type: ZSLDataset
   dataroot: datasets
-
+  
```
remove spaces.

adityac8 commented on options/tfVAEGAN/CUB.yml:
```diff
@@ -32,15 +32,15 @@ network:
 
   feedback:
-    lr: 0.00001
+    lr: 0.00001 
```
remove spaces.

adityac8 commented on train_images.py:
```diff
@@ -10,26 +10,26 @@
 import datasets.image_util as util
 import classifiers.classifier_images as classifier
-from config_images import opt
+from utils.logger import *
```
remove * imports.

adityac8 commented on train_images.py:
```diff
@@ -10,26 +10,26 @@
-    torch.cuda.manual_seed_all(opt.manualSeed)
-cudnn.benchmark = True
-if torch.cuda.is_available() and not opt.cuda:
+opt = parse_options()
```
Give option from `argparse` to pass a `yml` file.

adityac8 commented on train_images.py:
```diff
@@ -10,26 +10,26 @@
-if torch.cuda.is_available() and not opt.cuda:
+opt = parse_options()
+
+if opt['manual_seed'] is None:
```
Random seed is always passed from yml.

adityac8 commented on train_images.py:
```diff
@@ -67,14 +67,23 @@
+os.makedirs(opt['log'],exist_ok=True)
+
+# initialize loggers
+# logger, tb_logger = init_loggers(opt)
```
Why is it commented?

adityac8 commented on train_images.py:
```diff
@@ -105,71 +114,72 @@ def generate_syn_feature(generator,classes, attribute,num,netF=None,netDec=None)
-optimizerF         = optim.Adam(netF.parameters(), lr=opt.feed_lr, betas=(opt.beta1, 0.999))
-optimizerDec       = optim.Adam(netDec.parameters(), lr=opt.dec_lr, betas=(opt.beta1, 0.999))
+optimizer          = optim.Adam(netE.parameters(), lr=opt["network"]["gan"]["lr"])
+optimizerD         = optim.Adam(netD.parameters(), lr=opt["network"]["gan"]["lr"],betas=(0.5, 0.999))
```
Add betas as a list of two elements in yml file.

adityac8 commented on train_images.py:
```diff
@@ -297,10 +307,12 @@ def calc_gradient_penalty(netD,real_data, fake_data, input_att):
-print('Dataset', opt.dataset)
+# logger.info('End of training.')
+
+print('Dataset', opt["datasets"]["name"])
```
Remove print and add logger.

adityac8 commented on utils/logger.py:
```diff
@@ -0,0 +1,191 @@
+    # logger.info(get_env_info())
+    logger.info(dict2str(opt))
+
+    # initialize wandb logger before tensorboard logger to allow proper sync:
```
Remove extra lines.

adityac8 commented on utils/logger.py:
```diff
@@ -0,0 +1,191 @@
+    return logger, tb_logger
+
+# @master_only
+def init_tb_logger(log_dir):
```
Remove extra lines.

adityac8 commented on utils/logger.py:
```diff
@@ -0,0 +1,191 @@
+
+
+# @master_only
+def init_wandb_logger(opt):
```
Remove extra lines.

adityac8 commented on utils/logger.py:
```diff
@@ -0,0 +1,191 @@
+    import torch
+    import torchvision
+
+    # from basicsr.version import __version__
```
Remove extra lines.

adityac8 commented on utils/options.py:
```diff
@@ -0,0 +1,95 @@
+    opt = parse(args.opt)
+
+    # random seed
+    seed = opt.get('manual_seed')
```
Remove extra lines.

adityac8 commented on utils/options.py:
```diff
@@ -0,0 +1,95 @@
+
+def parse_options():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-opt', type=str, required=True, help='Path to option YAML file.')
```
Give this option in train.py

animesh-007 committed:
Made suggested changes

Removed Unnecessary functions from logger

Removed dict2str

Made suggested changes and removed extra functions

adityac8 commented on train_images.py:
```diff
@@ -10,36 +8,48 @@
+torch.manual_seed(opt['manual_seed'])
+if torch.cuda.is_available():
+    cuda = True
+if cuda:
```
remove this line.

adityac8 commented on networks/TFVAEGAN_model.py:
```diff
@@ -20,8 +19,9 @@ class Encoder(nn.Module):
-        layer_sizes = opt.encoder_layer_sizes
-        latent_size = opt.latent_size
+        layer_sizes = opt["network"]["gan"]["layer_sizes"]
+        layer_sizes[0] = opt["network"]["gan"]["res_size"]
```
This is confusing.

adityac8 commented on options/tfVAEGAN/CUB.yml:
```diff
@@ -7,23 +7,28 @@ datasets:
   type: ZSLDataset
   dataroot: datasets
-
+ 
```
Remove space.

adityac8 commented on options/tfVAEGAN/CUB.yml:
```diff
@@ -32,7 +37,8 @@ network:
 
   decoder:
-    lr: 0.001
+    layer_sizes: [4096, 8192]
+    lr: 0.0001
```
Change order.

adityac8 commented:
See unresolved issues from before also.

adityac8 commented on options/tfVAEGAN/CUB.yml:
```diff
@@ -7,23 +7,28 @@ datasets:
 
+optimizer:
+  adam:
+    betas: [0.5, 0.999]
```
Include it under `train`.

animesh-007 committed:
updated code

animesh-007 committed:
Update train_images.py

animesh-007 committed:
Update options.py
